contracts/AelinUpFrontDeal.sol

```diff
@@ -257,43 +262,48 @@ contract AelinUpFrontDeal is AelinERC20, MinimalProxyFactory, IAelinUpFrontDeal
+                    (((poolSharesPerUser[msg.sender] * _underlyingDealTokenTotal) / totalPoolShares) *
+                        (BASE - AELIN_FEE - dealData.sponsorFee)) /
+                    BASE;
                 poolSharesPerUser[msg.sender] = 0;
 
                 // refund any purchase tokens that got deallocated
                 uint256 purchasingRefund = purchaseTokensPerUser[msg.sender] -
                     ((purchaseTokensPerUser[msg.sender] * _underlyingDealTokenTotal) / totalPoolShares);
-                purchaseTokensPerUser[msg.sender] = 0;
 
-                uint256 precisionAdjustedRefund = purchasingRefund > IERC20(_purchaseToken).balanceOf(address(this))
+                precisionAdjustedRefund = purchasingRefund > IERC20(_purchaseToken).balanceOf(address(this))
                     ? IERC20(_purchaseToken).balanceOf(address(this))
                     : purchasingRefund;
 
-                // mint deal tokens and transfer purchase token refund
-                _mint(msg.sender, adjustedDealTokensForUser);
+                // Transfer purchase token refund
                 IERC20(_purchaseToken).safeTransfer(msg.sender, precisionAdjustedRefund);
-
-                emit ClaimDealTokens(msg.sender, adjustedDealTokensForUser, precisionAdjustedRefund);
             } else {
                 // mint deal tokens when there is no deallocation
-                uint256 adjustedDealTokensForUser = ((BASE - AELIN_FEE - dealData.sponsorFee) *
-                    poolSharesPerUser[msg.sender]) / BASE;
-                poolSharesPerUser[msg.sender] = 0;
-                purchaseTokensPerUser[msg.sender] = 0;
-                _mint(msg.sender, adjustedDealTokensForUser);
-                emit ClaimDealTokens(msg.sender, adjustedDealTokensForUser, 0);
+                adjustedShareAmountForUser =
+                    ((BASE - AELIN_FEE - dealData.sponsorFee) * poolSharesPerUser[msg.sender]) /
+                    BASE;
             }
+            poolSharesPerUser[msg.sender] = 0;
+            purchaseTokensPerUser[msg.sender] = 0;
+
+            // mint vesting token
+            _mint(msg.sender, tokenCount);
+            tokenDetails[tokenCount] = TokenDetails(adjustedShareAmountForUser, block.timestamp);
+            emit ClaimVestingToken(msg.sender, tokenCount, adjustedShareAmountForUser, precisionAdjustedRefund);
+            tokenCount += 1;
         } else {
             // Claim Refund
-            uint256 currentBalance = purchaseTokensPerUser[msg.sender];
+            uint256 refundAmount = purchaseTokensPerUser[msg.sender];
             purchaseTokensPerUser[msg.sender] = 0;
             poolSharesPerUser[msg.sender] = 0;
-            IERC20(_purchaseToken).safeTransfer(msg.sender, currentBalance);
-            emit ClaimDealTokens(msg.sender, 0, currentBalance);
+            IERC20(_purchaseToken).safeTransfer(msg.sender, refundAmount);
+            emit ClaimRefund(msg.sender, refundAmount);
```

Comment:
same here. we should not be renaming things as it will make the subgraph much more confusing. Again, we can rename this in the UI if we want to help users

Reply:
the current event tracks both amountPurchasingReturned and amountMinted which should be ok

Reply:
maybe I create a new event `CreateVestingToken` that I emit in `_createVestingToken`. Not sure what to do with `ClaimRefund`. Creating a new event just to show the refund is a bit overkill.